### DISCUSSION TITLE
Collapsible System Tray should be a built-in feature in Cinnamon

### DISCUSSION THREAD

**Discussion by moloch1994** [linuxmint/discussions#367] [Ideas]
On Windows it's been possible to hide tray icons since Windows XP, first with horizontal expansion:
![showtrayicons-xp](https://github.com/linuxmint/discussions/assets/68343381/718a4803-a76f-4cea-9bc8-909ada22582f)
Then, starting with Windows 7 with a pop-up:
![img01](https://github.com/linuxmint/discussions/assets/68343381/5685a558-be96-4569-abd5-774719e0bdb1)

On Linux Mint Cinnamon there is no such integral feature, so people have been resorting to custom applets. There are currently 3 "working" applets that are said to perform this function:
1. [Collapsible Systray](https://cinnamon-spices.linuxmint.com/applets/view/265)
2. [Drawer](https://cinnamon-spices.linuxmint.com/applets/view/169)
3. [Hideable Applets](https://cinnamon-spices.linuxmint.com/applets/view/293)

I will describe problems with each of them:
1. Collapsible Systray worked perfectly fine on Mint versions 20.3 and older. With the only "flaw" being that it only hides the apps' tray icons and not other applets. Not a big issue, if an issue at all. However, now there's a bug that when you click on the "expand" button, it instead clicks on one of the hidden applets, therefore making the whole tray unexpandable and thus unusable. The only way to fix this is to either restart Cinnamon or expand it in some other way, for example with a keyboard shortcut or by enabling "expand on hover". Essentially, this applet is broken and is not likely to be fixed, as we see little to no activity from the creator/maintainer.
2. Drawer is a more capable applet as it can hide anything on the panel. However, some apps just wouldn't hide (e.g. uGet) and will actually block the applet from expanding. In addition, I found it very unreliable, as it would occasionally just stop working for no apparent reason or fail to hide certain apps (e.g. qBitTorrent) upon login, requiring you to relaunch those apps.
3. I've seen people refer to Hideable Applets as an alternative, but it really works very differently. You need to click on its icon to summon a pop-up, and then click on the applet in the pop-up to unhide it, and only then you can interact with the applet. After that you have to click on the icon again and select the applet you wanna hide again to hide it. Meaning, instead of 2 clicks, you end up performing 5. This is very clunky and inefficient.

Whatever the status of these applets is right now, we shouldn't rely on volunteers to maintain them. I view this feature as essential for a modern desktop experience, and it is present not just in Windows, but also in other desktops, such as KDE:
![tray](https://github.com/linuxmint/discussions/assets/68343381/1e958d9c-5082-4b64-948e-8307f7d5716e)
Deepin:
![Screenshot from 2024-03-18 14-27-17](https://github.com/linuxmint/discussions/assets/68343381/6fc07fa2-5c64-461d-924d-4e1e16d63fe9)
UKUI:
![Screenshot from 2024-03-18 14-28-39](https://github.com/linuxmint/discussions/assets/68343381/64eb8133-df9a-4941-bedb-c8f728b7e7b2)
[Xfce](https://www.addictivetips.com/ubuntu-linux-tips/hide-apps-from-the-system-tray-in-xfce4/) and others. With, perhaps, the notable exception being Gnome, which is intentionally deviating from traditional paradigms.

Without this feature my tray ends up occupying nearly half of my panel (on a laptop with 1366x768 resolution), forcing me to only use the modern Grouped Window List layout, which I'm not a fan of. I implore the developers to please make this feature an integral part of Cinnamon instead of forcing us to use unreliable custom applets.

**Comment by mfreeman72** [comment]
I agree. I've been using Drawer until Mint 22 came out. It worked perfectly for everything I used up through 21.3. Since the upgrade to 22, Drawer seems to work well with fewer tray icons for some reason, causing Drawer to malfunction frequently. It would be good to have an officially sanctioned option, since most 3rd party applets in Cinnamon Spices seem to be completely abandoned by their author and broken in one way or another (a frustrating discussion for another thread).

**Comment by EnthusiasticRepositoryUser** [comment]
I agree so much. Coming from Windows, I've been trying out different desktop environments, and overall Cinnamon is the one I love the most with KDE Plasma a close second. The one thing I adore about Cinnamon is the corner bar, that works and looks like in Windows. You don't get that anywhere else, not in that style. The one thing I am absolutely sorely missing is a collapsable systray icon. 

I think it would be an awesome addition, considering the corner bar from Windows is here, the collapsable systray icon should be added too.

**Comment by cplittlfield** [comment]
On the forums, we see a lot of these requests to make Linux Mint behave exactly as Windows does.

Very few people seem interested in learning a new way of doing things, which I personally think is a pity.  My personal view is that if I wanted precise Windows-like behavior, I would have stayed on Windows.  For me, the Cinnamon desktop environment is enough like Windows to be comfortable, but enough different to be refreshing.  Now, when I am forced to use Win 11 on my dual-boot machine at the office, I deeply regret not having switched to Linux years earlier.

**Reply by moloch1994** [reply]
> Very few people seem interested in learning a new way of doing things, which I personally think is a pity. 

I initially thought you were replying to my post about the clock format, but the same criticism applies here. Having no collapsible systray is not new. This is what Windows 95 did.

**Reply by cplittlfield** [reply]
Instead of wasting energy berating me, I see three possible paths forward for you.  One would be too stop complaining that the Cinnamon DE is not confirming to your will,  and simply adapt.   The second is to write a systray applet that does what you want.  The third would be to find a distro that does what you want in every detail. Peace and blessings, regardless of the path you choose.

**Reply by moloch1994** [reply]
You posted a criticism of my post, but I pointed out how your argument was flawed. It's just plain arrogant and condescending to act like everybody else is rigid and simply can't move past the Windows paradigm, when the very distro is enforcing that paradigm.

Your "three paths" are yet again rehashing the common toxic reply constantly seen in the Linux community: "You don't like it? Do it yourself." This logic means there's no place for constructive criticism and contributions. I've actually donated money to the Linux Mint project and I simply respectfully ask the developers to consider my proposal. I don't demand it of them or criticize them for not implementing it, I'm simply offering a suggestion. For this I should be hearing that I'm "not interested in learning new ways of doing things," implying that I'm narrow-minded and rigid or that I should become a developer and write my own applet that I would then have to maintain for eternity? Yeah, maybe we should just all stop proposing ideas or reporting bugs, how much better would that be then.

**Reply by EnthusiasticRepositoryUser** [reply]
Wanting quality of life and good features implemented, that happen to be on or come from Windows is not refusing to learn a new system. If Windows does something good, adopt it. The collapsable systray is undeniably a good thing to have. Not implementing it only because of some sene of gatekeeping is just not very reasonable.

**Comment by ghost** [comment]
3/4 of my task bar is filled on a 1366 px wide display, and a solid 2 inches of my 15 inch display is taken up solely by system and app indicators.

If a collapsible tray is for some reason to much to ask for then could at least the WiFi, sound, and notifications indicators be combined into a single pop up or something via a single setting or something.

And I have used the collapsible tray applets in the past and if they somehow managed to work correctly all 3 caused stability problems.

and quite frankly if I wanted to hear "use a third party add on that maybe works, because we are not adding that" I would use Gnome and Gnome is very user hostile so that is not a image you want to inspire of this project.

If nothing else the "Modern" panel is more like windows XP crossed with windows vista/7 and as mentioned above they both had collapsible indicators built in, you could even hide the WiFi, battery, and Bluetooth indicators if you did not care to see them most of the time to save room for other things you actually want to use.

I would be happy to leave this issue be If their is a technical reason for basically shutting down the whole conversation.

**Reply by mfreeman72** [reply]
You can always install the Drawer applet. It seems to be working fine now. At least on my system.

**Reply by ghost** [reply]
I'll take a look at that.

**Reply by ghost** [reply]
It did not show any indicators, and I tried the trick of removing the built in xapp indicators applet.

So unless I'm supposed to do something else to make it work, it was worse then the default experience.

I do not see a good reason to not add a "auto hide indicators" option to the default first party applet.

**Reply by mfreeman72** [reply]
It doesn't work like that.

First, DO NOT remove the "Xapp Status Applet"!!! Only remove the "System Tray" applet, and *only* if there are glitches with Drawer (such as certain icons disappearing or the Drawer applet refusing to open or close). But always start with both "Xapp Status Applet" and "System Tray" applets installed and only remove "System Tray" if glitches occur.

Second, here's how you use the Drawer applet:

- In System Settings -> Applets, on the Manage tab: Select "Drawer (show/hide applets)".
- Click the + button at the bottom of the window. This will add the applet to the panel.
- Right-click the panel and turn on "Panel edit mode".
- Drag the Drawer applet (it should look like a small arrow or triangle) to the right side of the applets you want to hide.
- Right-click the panel and turn off "Panel edit mode".

Try that and see if that works for you. It works perfectly for me.

**Reply by moloch1994** [reply]
You see, the fact that a user has to go through all this, it's kinda the reason there needs to be a native implementation. It's confusing and you have to figure it out through trial and error.

**Reply by ghost** [reply]
After I read your tips, because the applet it's self has no instructions on how to make it work.
![Screenshot from 2025-06-25 10-12-18](https://github.com/user-attachments/assets/8564bd33-6a36-4565-8965-7bc7a3a1297e)
![Screenshot from 2025-06-25 10-12-50](https://github.com/user-attachments/assets/b06d7793-f381-4288-94c9-6f25ec494b43)
It proceded to hide my indicators and not show anything, and considering the statement "First, DO NOT remove the "Xapp Status Applet"!!! Only remove the "System Tray" applet, and only if there are glitches with Drawer (such as certain icons disappearing or the Drawer applet refusing to open or close). But always start with both "Xapp Status Applet" and "System Tray" applets installed and only remove "System Tray" if glitches occur." I suspect It will break in the future even If I fixed it today, and after I removed drawer I still needed to restart cinnamon to bring the hidden stuff back.

Unless the bottom picture is the instructions, and their is definitely no mention of troubleshooting tips so I used it the way it was clearly intended to be used. add it to the panel, and infer from the picture something about edit mode and assume it is broken if it doesn't work, like all computer users do.

In short drawer has improved nothing but it's appearance, and is broken just like all of it's forebears.

**Reply by mfreeman72** [reply]
So, I think I discovered the problem. Linux Mint's "Xapp Status Applet" and "System Tray" applets both do the same job -- display status icons. Unfortunately, Linux being what it is, there have been many different methods used over the years for displaying these status icons, each invented by the makers of the different desktop environments. Some of the older methods have been deprecated in recent years. But older programs (such as uGet) still use these methods. "Xapp Status Applet" doesn't support the older methods of displaying these icons. "System Tray", which is Linux Mint's older status icon display applet, does still support them. This is why Linux Mint still includes both by default. However, "System Tray" doesn't look out for when it has been hidden by another applet, such as Drawer. So once applets have been hidden by Drawer, and Drawer tries to display them again, they don't know how to re-display their icons. So they vanish.

This means that any method used for hiding icons on the panel, official or unofficial, will always cause these older status icons to disappear.

So, you have four options to make this work:

1. Move Drawer, and any icons you want Drawer to hide, to the left of "System Tray", so that the older status icons remain always displayed.
2. Find newer alternatives to your older programs.
3. Send a bug report to developers of the older programs and ask that they update them to support newer status icon methods.
4. Send a bug report to Linux Mint and ask that they make their "System Tray" applet work better with applets that hide other applets on the panel.

**Reply by mfreeman72** [reply]
So, to summarize: No. Drawer is not broken. Linux Mint's "System Tray" is. That's where the problem is occurring, and will always occur anytime the "System Tray" applet is hidden, no matter which applet is doing the hiding. Even if it's a "native" one.

**Reply by mfreeman72** [reply]
Another issue to consider: uGet (and probably any program still using deprecated status icon methods) is no longer being developed, according to its Github page (https://github.com/ugetdm/uget), and the last version I can find (2.2.3) is from 5 years ago. That's ancient, in programming and network security terms. It's really not recommended to use network-facing programs that are no longer being developed or receiving security updates, as they could pose a security risk to your system. You are, of course, free to use whatever you want to use, if you personally don't care about such things. However, it would both help you solve this applet hiding issue, as well as improve your online security, to find a different download manager. There are several available on Software Manager (just type "download manager" into the search bar).

**Comment by mfreeman72** [comment]
> 2. Drawer is a more capable applet as it can hide anything on the panel. However, some apps just wouldn't hide (e.g. uGet) and will actually block the applet from expanding. In addition, I found it very unreliable, as it would occasionally just stop working for no apparent reason or fail to hide certain apps (e.g. qBitTorrent) upon login, requiring you to relaunch those apps.

In System Settings -> Applets, do you have both "System Tray" and "XApp Status Applet" active on the panel? I've found that things work better with Drawer if you disable "System Tray" and only have "XApp Status Applet" running.

**Reply by moloch1994** [reply]
I've tried it every which way. uGet simply does not work with Drawer, I have to just disable the tray icon in the settings. qBitTorrent normally works, but the bug I described happens sometimes. For example, if I want it to launch on startup, I have to set a delay, to ensure it launches after everything else. That way it works.

For the most part drawer works well, but it's those few bugs and the potential of more appearing in the future that makes me want a native applet.

**Reply by mfreeman72** [reply]
I took a look at uGet. It appears to be using an old, outdated method of displaying a status icon in the system tray. So in this respect, "the potential of more appearing in the future" isn't an issue. uGet is an old program. Any new programs use the newer, better-supported methods. It's the reason Linux Mint includes both "Xapp Status Applet" and "System Tray" applets. They both essentially serve the same purpose. However, "Xapp Status Applet" doesn't support the older, deprecated method of displaying a status icon. System Tray (which is their original applet for doing this job) does, but it's not compatible with newer applets like Drawer. So you're kind of stuck if you use older programs that refuse to update their status icon display methods, I'm sorry to say. You may want to contact the uGet developers, if they're still around.

**Reply by mfreeman72** [reply]
I also found a workaround for you:

- With uGet running (so that its icon is displayed on the panel), add Drawer to the panel, but keep it to the left of uGet's icon.
- Right-click on the panel and turn on "Panel edit mode".
- Move all applets you want hidden by Drawer to the left of Drawer's arrow icon, *EXCEPT* uGet. Keep that to the right of Drawer.
- Right-click on the panel and turn off "Panel edit mode".

This will make uGet (and any programs that use the older status icon display method) stay outside of the Drawer applet's hidden icons and always visible. Yeah, there will be some icons you can't hide with Drawer, but it'll hide some and maintain all of your programs' functions. The issue, I believe, is that the "System Tray" applet doesn't know how to re-display its icons once they've been hidden, so they just disappear and never return. So keeping them outside of Drawer prevents them from disappearing.

**Reply by ghost** [reply]
So we are going with the classic "this is not a supported use case, use something else" gnome response?
It seems that cinnamon's implementation even if it is split into two applets works as is, so add auto-hide functionality to both of them.
Or just add the "old standard" support to xapp, and add a way to auto-hide indicators if possible and that would also solve the problem permanently, and reduce how many system applets need to be maintained.

Or as a third option just add auto-hide functionality to xapp, and leave  the system tray as is.
I missed your response before this one.

**Reply by mfreeman72** [reply]
@fosslover306 I think you've misunderstood me. I'm not telling you to use something else. You're free to use whatever you want. I'm just saying that, due to Linux's fragmentation, you will occasionally find little issues like this. Especially when older programs use methods that are no longer fully supported.

I'm not a Linux Mint developer, so I cannot speak for them. They may be able to do exactly what you're asking for. But I would recommend that you make a new, more specific bug report, so that this issue is more directly seen and understood by them. I'm just providing you with some alternatives to remedy the situation. You're welcome to send in a bug report for their "System Tray" applet, which is the root cause of this problem. Or you can use the workaround I mentioned. Or you can use a different program. Or you could even create your own applet to solve the problem, if you have the programming skills to do so. These are all just options that you have. They may not even be the only ones. I'm just trying to help you find ways to accomplish what you're trying to do, within the current limitations of the system.

**Reply by ghost** [reply]
Apparently the "systray" or "system tray" does not exist, and their is no about page for systray in the applet manager, and I also searched the linux mint repositories list and it is not coming up, so where do I report this bug, or feature request to?
![Screenshot from 2025-06-25 11-43-43](https://github.com/user-attachments/assets/639725de-4f7e-4df3-bd8a-5921b20e6c17)

**Reply by mfreeman72** [reply]
@fosslover306 It exists, but it doesn't have its own github page. Since it's one of Cinnamon's included applets and not a separate Spices download, my guess is that you'll probably need to send bug reports to the Cinnamon issue tracker: https://github.com/linuxmint/cinnamon/issues

Any of Cinnamon's built-in applets (indicated in the applet settings with a lock icon in your image) are like this. So you'll have to send reports for any of those applets to the Cinnamon github issue tracker as well.

**Reply by moloch1994** [reply]
> I also found a workaround for you:
> 
>     * With uGet running (so that its icon is displayed on the panel), add Drawer to the panel, but keep it to the left of uGet's icon.
> 
>     * Right-click on the panel and turn on "Panel edit mode".
> 
>     * Move all applets you want hidden by Drawer to the left of Drawer's arrow icon, _EXCEPT_ uGet. Keep that to the right of Drawer.
> 
>     * Right-click on the panel and turn off "Panel edit mode".
> 
> 
> This will make uGet (and any programs that use the older status icon display method) stay outside of the Drawer applet's hidden icons and always visible. Yeah, there will be some icons you can't hide with Drawer, but it'll hide some and maintain all of your programs' functions. The issue, I believe, is that the "System Tray" applet doesn't know how to re-display its icons once they've been hidden, so they just disappear and never return. So keeping them outside of Drawer prevents them from disappearing.

Yeah, I kinda figured that "workaround" myself, which isn't a workaround for me at all. Maybe it's ADD or something, but I can't stand seeing one icon not behave like the rest, it stands out like a sore thumb. I'd rather just disable the icon altogether.

But anyway, the old Collapsible Systray applet used to hide it perfectly and even added proper applet hover highlight at the time when it wasn't there by default on Mint. It really did equalize all tray icons for apps, even allowing the user to adjust spacing between them. It's a shame it broke down. But its existence shows that this can be done if the devs choose to.

**Reply by mfreeman72** [reply]
> Yeah, I kinda figured that "workaround" myself, which isn't a workaround for me at all. Maybe it's ADD or something, but I can't stand seeing one icon not behave like the rest, it stands out like a sore thumb. I'd rather just disable the icon altogether.
> 
> But anyway, the old Collapsible Systray applet used to hide it perfectly and even added proper applet hover highlight at the time when it wasn't there by default on Mint. It really did equalize all tray icons for apps, even allowing the user to adjust spacing between them. It's a shame it broke down. But its existence shows that this can be done if the devs choose to.

Again, I can't speak for the Mint devs, but personally, I wouldn't think the devs are going to choose to change their applet for something that has been deprecated and mainly exists programs that are outdated and have basically been killed off by their developers, and for something that is never seen by most people who use their operating system. They are a small team of developers, and have to place their priorities on things that make a difference for the most people and for forward-looking software developments. I'm sorry that a program you enjoy isn't working with the Drawer applet and isn't being supported by its developers anymore. It's a frustrating situation, I'm sure. All I can recommend is to try out other download managers (there are many of them in the Software Manager), and see if one of them can do the job well enough for you.

**Reply by moloch1994** [reply]
I don't understand why you're hyperfocused on uGet. I did mention that there are other minor issues. Plus, it's not like they need to reinvent the wheel. An applet already existed that worked perfectly with all apps. All that needs to be done is to fix the bug that was introduced with Mint 21 and maintain the applet officially. You're making it sound more complicated than it is.

**Reply by mfreeman72** [reply]
> I don't understand why you're hyperfocused on uGet. I did mention that there are other minor issues.

Because it's the only concrete example you gave. Just about anything I said about uGet should apply to any program that uses deprecated technologies that are no longer fully supported. If you can give me some more specific examples, I can address them as well, if you'd like.

> Plus, it's not like they need to reinvent the wheel. An applet already existed that worked perfectly with all apps. All that needs to be done is to fix the bug that was introduced with Mint 21 and maintain the applet officially. You're making it sound more complicated than it is.

This isn't exactly reinventing the wheel, and it's not complicated, and it's not a bug. Before, there were several different and conflicting ways to display status icons. Methods that have been less effective or more difficult to work with are being left behind, while better ways are becoming the standard. Once the standard is in place, it's the expectation that currently-supported programs follow the newer standards and let the old methods die off. Old programs that are no longer being developed should not be expected to function with newer standards, because doing so would create unnecessary work for little return on time invested. Linux Mint still allows for the use of outdated programs that use the older standard by keeping an applet around that works with it, but no one should expect that a tiny organization like Linux Mint's dev team make it a priority to keep a dying standard fully compatible with every new applet and option that's invented.

**Reply by moloch1994** [reply]
>Because it's the only concrete example you gave.

No, I also said this:

>as it would occasionally just stop working for no apparent reason or fail to hide certain apps (e.g. qBitTorrent)

Idk what more "concrete" examples you need, when most of the time the applet is failing randomly with no discernible pattern. Plus, as it's clear from the issues @fosslover306 experienced, it's not exactly easy to set up for some users.

>This isn't exactly reinventing the wheel, and it's not complicated, and it's not a bug. Before, there were several different and conflicting ways to display status icons. Methods that have been less effective or more difficult to work with are being left behind, while better ways are becoming the standard. Once the standard is in place, it's the expectation that currently-supported programs follow the newer standards and let the old methods die off. Old programs that are no longer being developed should not be expected to function with newer standards, because doing so would create unnecessary work for little return on time invested. Linux Mint still allows for the use of outdated programs that use the older standard by keeping an applet around that works with it, but no one should expect that a tiny organization like Linux Mint's dev team make it a priority to keep a dying standard fully compatible with every new applet and option that's invented.

Again, I don't know what any of this has to do with an already existing applet that already supports all formats. I've mentioned this three times:

>Collapsible Systray worked perfectly fine on Mint versions 20.3 and older. However, now there's a bug that when you click on the "expand" button, it instead clicks on one of the hidden applets, therefore making the whole tray unexpandable and thus unusable. 

>But anyway, the old Collapsible Systray applet used to hide it perfectly.

>An applet already existed that worked perfectly with all apps.

To reiterate, we already have an applet that works with all formats. The bug it's suffering from has nothing to do with any specific application, therefore is probably unrelated to the old standards you keep referring to. All that's needed to do is to take that applet, fix it, and maintain it as official.

**Comment by Krezzin** [comment]
+1. Mint lacks a lot in terms of UX customization. From collapsing "Quick Launch" panel, to the collapsable system tray you can organize however you want by drag-n-drop. Dunno why it doesn't have that.

**Reply by mfreeman72** [reply]
> +1. Mint lacks a lot in terms of UX customization. From collapsing "Quick Launch" panel, to the collapsable system tray you can organize however you want by drag-n-drop. Dunno why it doesn't have that.

It *does* have those things as 3rd party applets. That's what this whole discussion is about.

**Reply by ghost** [reply]
You mean broken and hard to use  collapsible system tray applets? Their are plenty of those working ones not so much.

**Reply by mfreeman72** [reply]
The Drawer applet works flawlessly for me, although you have to make sure the outdated "System Tray" applet that comes with Linux Mint (as opposed to the "Xapp Status Applet" that also comes with Mint) is either disabled, or kept outside of the hidden applet area, since it's not compatible with it, and I don't think the Linux Mint team is actively developing it much anymore. I haven't tested the others recently, but I'll do that and report back, if you're interested.

**Reply by ghost** [reply]
Did drawer actually add instructions to make it work? If you check my screen shots taken when a posted them they did not exsist, and someone else said not to disable anything to make it work, it could cause bugs, and only if it causes bugs remove the default systray.

I'm not going to screw around with my system idicators by playing logic puzzle games.

**Reply by mfreeman72** [reply]
> Did drawer actually add instructions to make it work? If you check my screen shots taken when a posted them they did not exsist, and someone else said not to disable anything to make it work, it could cause bugs, and only if it causes bugs remove the default systray.
> 
> I'm not going to screw around with my system idicators by playing logic puzzle games.

It's not a puzzle game. The "System Tray" applet isn't compatible with Drawer, so if you want to use Drawer, you have to keep it outside of Drawer's hidable area. It's incredibly simple. And since the "System Tray" applet is only there for backward compatibility with extremely old software that uses a deprecated System Tray protocol, then if you're not using old, outdated software, then you can safely disable it, and you probably have never seen it actually display anything anyway in that case. That's it. Nothing scary or confusing about it.

**Reply by mfreeman72** [reply]
Also, that "someone else" who said "not to disable anything to make it work" was me, and that's not *quite* what I said. I apologize if I was confusing. If you need instructions to make Drawer work properly, here they are:

1. Install Drawer (go to System Settings -> Applets, click the "Download" tab, find Drawer in the list, and click the download button to the right of the Drawer applet)
2. While still in System Settings -> Applets, click the "Manage" tab.
3. Click to highlight Drawer.
4. Click the + button at the bottom of the window. This will add Drawer to your main panel.
5. Right-click on your panel and activate "Panel edit mode".
6. Organize all of the panel applets that you want hidden into the left-most side of the panel grouping area you are using (panel grouping area being either the left, center or right areas that are highlighed as red, green, or blue when in Panel Edit Mode).
7. Move the Drawer applet arrow to the immediate right of the applets you want to hide, keeping any applets you don't want hidden moved to the right of Drawer.
8. Make sure that the "System Tray" applet (*NOT* "Xapp Status Applet") is *not* to the left of the Drawer applet, otherwise it may cause panel icons to disappear. To do this, you may need to run a program that uses this specific older System Tray protocol to make it visible. If you have no older programs that use the "System Tray" applet, you can safely disable it. If you're not sure of the difference between the two, I recommend just disabling the "System Tray" applet until you find that a program's tray icon isn't displaying. At that point, you can re-enable the "System Tray" applet, put the panel into Panel Edit Mode, and move that specific program's icon to the right of Drawer.
9. Once everything is organized the way you want it, right click on the panel and turn off Panel Edit Mode.
10. You may click the "Configure" icon next to Drawer in System Settings -> Applets -> "Manage" tab to explore and adjust the applet's settings. If you make a mistake, you can always click the menu button at the top-right and select "Reset to defaults" to get all the original settings back. When done, you can close all settings windows.

As I was trying to be as detailed as possible, these instructions look more complex than they actually are. They are mostly just generic applet management instructions.

**Comment by Krezzin** [comment]
I tried all available options, from Drawer, to AppLaunch, App Launch is the only kinda Quick Launch alternative, Drawer works but lacks an option to move the icons. It's still something. Although its always good to have this part of the OS. Doubt its coming soon since it's not been in Mint for years.

**Reply by mfreeman72** [reply]
I see. So the biggest sticking point for you is that you want to be able to move things around within the "Xapp Status Applet"/"System Tray" applets themselves, is that correct? Because you can move applets around anywhere you want on the panel, but within these System Tray applets, the icons generally stay in their default order. If so, that's a different issue than this thread is about, I think. I'm not sure of the specifics, but I think the issue with what I think you want to accomplish is that Cinnamon is trying to accommodate a variety of different System Tray protocols from different desktop environments for compatibility, and changing the order might break something in compatibility, so it has to be locked down. I could be wrong, and I'd love to hear dev feedback on that. I know I've requested a feature for the System Tray to make the dropdown menu for these icons match the main panel applet menu theme. Mint used to have this as an experimental feature years ago (Mint 17? 18?), and it worked beautifully for me, but it was said to be extremely problematic for some users, depending on the compatibility of the software they used, so they removed that. I think your request might have a similar issue, if that makes sense.

**Reply by ghost** [reply]
Those are some nice instructions, which would be very nice to read in the applets main settings page, or even the applets info page, and I hate to say it but if the required instructions to get something to work are not listed anywhere but a random github discussion, and read like a role playing games context setup scene to get all the players on the dungeon masters vision of the adventure, maybe it is to complex to setup, or is to fragile to work reliably, and yes I'm going to leave out the stuff about installing drawer. After that point it still sounds like I will break it if I click or look at it wrong or something, and I still have to change the programs I use that may or may-not have problems just to use something that does something that was built into windows xp 20 years ago, and worked flawlessly, and actually did work in one applet but it was abandoned so it broke, but it could still handle both new and old ways to do things so claiming that it's my choice of setup or programs I use and it can't be done is just making excuses. 

Either the default systray gets collapsible or not at this point, and whoever maintains drawer needs to add some actual usage instructions so one can actually use it properly, and provide answers when stuff don't work.

"Right-click on your panel and activate "Panel edit mode".
    Organize all of the panel applets that you want hidden into the left-most side of the panel grouping area you are using (panel grouping area being either the left, center or right areas that are highlighed as red, green, or blue when in Panel Edit Mode).
    Move the Drawer applet arrow to the immediate right of the applets you want to hide, keeping any applets you don't want hidden moved to the right of Drawer.
    Make sure that the "System Tray" applet (NOT "Xapp Status Applet") is not to the left of the Drawer applet, otherwise it may cause panel icons to disappear. To do this, you may need to run a program that uses this specific older System Tray protocol to make it visible. If you have no older programs that use the "System Tray" applet, you can safely disable it. If you're not sure of the difference between the two, I recommend just disabling the "System Tray" applet until you find that a program's tray icon isn't displaying. At that point, you can re-enable the "System Tray" applet, put the panel into Panel Edit Mode, and move that specific program's icon to the right of Drawer.
    Once everything is organized the way you want it, right click on the panel and turn off Panel Edit Mode.
    You may click the "Configure" icon next to Drawer in System Settings -> Applets -> "Manage" tab to explore and adjust the applet's settings. If you make a mistake, you can always click the menu button at the top-right and select "Reset to defaults" to get all the original settings back. When done, you can close all settings windows."

**Comment by emmegamma** [comment]
+1. While I'm a fan of Drawer, I do agree that, no matter what, Cinnamon needs some default way of hiding/reducing the number of icons. It's quite weird that it is not available, while there's TWO applets to manage system tray icons, one of which appears to be there only for obsolete apps but at the same time breaking other useful applets (Drawer)... crazy! And I only found out thanks to this thread, otherwise I might have migrated to some other DE/distro.

The easy and practical solution would be to not have System Tray in the panel by default (but installed and ready as "System tray (legacy)" or something), and then have Drawer instead. It's also possible to remove System Tray and substitute XAppStatusApplet (or both? not sure which one it resembles more in that aspect) with Collapsible System Tray, but that would require to test it and maintain it which might be quite some extra work for the devs.. not sure it's desirable.

**Comment by Secret-chest** [comment]
FYI, MATE has a drawer applet that works a bit differently because it creates a subpanel.

**Comment by BeatLink** [comment]
@clefebvre Hi Clem. I'm wondering if your team would consider implementing this before the next release. Its the third most upvoted feature in the entire discussions thread.

**Comment by Jopp-gh** [comment]
Wow, a lot of window switchers!
I come from MacOs where nobody ever cared of collapsing sys treats. 
Some 3rd party developer made an app called Butler, if I remember well.   

But back to the sys tray: for this kind of requests, Mint's splices should fill the gap. I recommend: 
https://cinnamon-spices.linuxmint.com/applets/view/344

Finally: also theme creators could add features like this with pure css, if Mint would unlock this kind of features. 

Linux is a lot of DIY, bc there isn't the manpower.

However this isn't all bad: because of this, Linux is lean and highly configurable. Configurable at least, for those proficient with command line prompts: that's not unfair, that's basically Linux.

**Reply by emmegamma** [reply]
You seem to be confusing the systray with an app launcher/menu.

**Reply by Jopp-gh** [reply]
No.
You can add terminal commands as well as applications to the suggested app launcher. This way you get rid of the clutter in your sys tray. Many applets / desklets are easily replaceable with commands. Some bash scripting may be useful.

**Reply by emmegamma** [reply]
I see what you mean, since there's some confusion in this topic between hiding applets and hiding systray icons, but you should really look up what a systray is (and how you won't reduce the clutter in it even if you removed _all_ of your applets)

**Reply by moloch1994** [reply]
>  I come from MacOs where nobody ever cared of collapsing sys treats.

Because on MacOS there is no taskbar (aka "window list") on the panel. I for one cannot stand the dock and prefer the layout I've been using (whenever possible) since my first OS, Windows 98.

**Reply by Jopp-gh** [reply]
@emmegamma 
The sys tray is just a windows paradigm. I don't doubt in this concept, but you can take many "routes to the top of a mountain". Lets try something else? 😊 

@moloch1994 
you'll laugh, the first thing I did on Mint after switching from MacOs (and trying many more distros before this step), was to remove the dock. 

Linux allowed me to configure the desktop, even the main UI and many more aspects to my liking, it was a new, "mint" fresh start to me.

**Reply by emmegamma** [reply]
> @emmegamma The sys tray is just a windows paradigm. I don't doubt in this concept, but you can take many "routes to the top of a mountain". Lets try something else? 😊

I think this deviates quite a lot from the original purpose: the systray is a recurrent topic in Linux forums as one thing that nobody wants but most people need (examples of apps that use it: ProtonVPN/NordVPN, Nvidia, syncthingtray, Dropbox, but even Mint's default Bluetooth stack)
Then, we need to add applets, which at the moment take one slot each on the panel (even Sticky Notes puts one).

Different approaches are welcome - Gnome, for instance, did a quite good job IMHO with the QuickSettings applet (many buttons/settings at hand, including those from extensions, but only 'active' icons in the panel). I think that's more similar to MacOS's approach but I don't really know.

A 'something else' approach would be welcome, I'd be up for something Gnome-like but I think that hiding icons is just the easiest to implement.
What seems pretty clear, from the upvotes, is that many people have issues with the ever-growing number of icons in the panel, and no easy solution if not through 3rd party applets: that's the issue at hand, and 'don't use applets, use bash' doesn't really sound as a solution for everyone

**Reply by moloch1994** [reply]
> @moloch1994 you'll laugh, the first thing I did on Mint after switching from MacOs (and trying many more distros before this step), was to remove the dock.

Not sure what you're talking about. There is no dock on Cinnamon by default (there's an extension for it). If you mean that you removed the Windows-like taskbar, which is the applet called "Grouped Window List", I don't really know how you manage your windows without it. But that's really besides the point. The reason we need to hide tray icons is because there needs to be space for launchers and windows on the panel. On Mac the panel only holds tray icons and a menubar.

**Reply by BeatLink** [reply]
@emmegamma 

System tray icons serve a valuable purpose in modern operating systems. They help the user to monitor and control long running background programs without having an always open window. Software like filesystem synchronizers, backup tools, music players, chat applications, email applications, etc all benefit from having a system tray icon. Which is why not only windows supports it but almost every single Linux desktop environment from KDE to MATE to XFCE to LXQT. I'd even dare to say that a system tray is as fundamental to the desktop experience as an application launcher / start menu

**Reply by emmegamma** [reply]
@BeatLink I agree, I'm not sure if anything I said suggested different.. or maybe you wanted to tag the user that calls it a Windows-thing?

When I said "something that nobody wants" I meant from a developer's perspective - which I actually don't really know about, I'm guessing because over the years it seems to have caused quite a lot of headaches, different protocols, and quite too many blog posts to be something fun to work on :D
I remember using some DEs/WMs of the early 2000s where it would just appear in a corner of the screen, floating freely :D

**Reply by moloch1994** [reply]
I think most of these minimalist ideas like removing the tray, desktop icons, or even the log out button (as Gnome did recently) come from purely aesthetic considerations. In most cases, there's no practical purpose to removing a feature without proposing an alternative.
